LibreNMS/OS/Fabos.php

```diff
@@ -118,4 +125,50 @@ protected function fcSwitchModelToName($model)
 
         return $models[$model] ?? 'Unknown Brocade FC Switch';
     }
+
+    public function discoverTransceivers(): Collection
+    {
+        $portCollection = $this->getDevice()->ports()->get();
+        $sensorCollection = $this->getDevice()->sensors()->get();
+
+        $snmpData = SnmpQuery::hideMib()->mibs(['FA-EXT-MIB'])->enumStrings()->walk('FA-EXT-MIB::swConnUnitPortTable')->table(1);
+        $snmpData = SnmpQuery::hideMib()->mibs(['FCMGMT-MIB'])->enumStrings()->walk('FCMGMT-MIB::connUnitPortTable')->table(1, $snmpData);
+
+        $snmpData = array_shift($snmpData);
+
+        Log::info('Transceivers discovery started');
+
+        $data = $portCollection->map(function ($port) use ($sensorCollection) {
```

Comment:
Why are you doing this?  No other devices doe this.

Reply:
Instead of using a fake ifIndex, which is what seems to be happening here.
just use connUnitPortIndex (and connUnitPortUnitId)

Reply:
To answer your 1st question -> did not find any other way so far :D 
And for the 2nd : Not sure how using connUnitPortIndex will change much. The goal is to map a port, a sensor and a transceiver. Sensors are already discovered, and do have already the ifIndex of the matching port. So I kept this ID for the transceivers.

Reply:
now, I use connUnitPortIndex and renamed the variables accordingly. connUnitPortUnitId does not help, it is a string (HEX).